pkg/virt-handler/heartbeat/heartbeat.go

```diff
@@ -103,6 +103,16 @@ func (h *HeartBeat) labelNodeUnschedulable() (done chan struct{}) {
 	return done
 }
 
+func (h *HeartBeat) isMasterNodeWorker() bool {
+	node, err := h.clientset.Nodes().Get(context.Background(), h.host, metav1.GetOptions{})
```

Comment:
This is expensive. What is the use-case to have a `virt-handler` on the node but being `unshedulable`?

Reply:
For some reason when the master nodes were being set to schedulable then to unschedulable, the virt-handler was added but not removed. It was mentioned that we may not want to kill the running virt-handler because we don't want to orphan any VMIs that were running while the node was set to schedulable. This is basically a workaround that keeps the virt-handler running but still sets nodes to unschedulable when they need to be. 

If there is no reason however to not kill virt-handler, I can instead look into killing the virt-handler on unschedulable nodes after validating no VMIs are running on the nodes. What do you think?

Reply:
The placement of `virt-handler` should be controlled by `.infra.workloads`(I did a mistake in my previous comment). It is true that it is desirable to let the admin decide if killing the virt-handler is safe.

One question, when the node is put back to schedulable, what does the daemonset says (in terms of desired and existing)?

I admit I don't see a clean way to express: On this node, we should not allow scheduling VMs(but let other pods run here) so we can do xyz. For purposes of giving time to existing VMs to migrate off the node. Is there a use case for this anyway? But if anything I would try to think about how we allow this rather than doing some magic with `control-plane` role which imho might be the wrong assumption.

Reply:
Since there was an issue with kubevirtci not assigning the worker label to the `control-plane` nodes when they were schedulable, it caused many lanes to fail so we decided to change this PR to have the heartbeat only check for the presence of the 'NoSchedulable` taint. What do you think?

Reply:
This still doesn't make sense to me. I would really appreciate it if you explain to me "why we need this change". There are zero reasons to have Kubevirt set its internal schedulable to false if you have `NoSchedule` as the `NoSchedule` will cause VMs not to schedule there. Or am I missing anything?

Reply:
For whatever reason, when the control planes are set to schedule and then unschedulable, virt-handler is created but not stopped when the node is set to schedulable. Because of the virt-handler continuing to run, the heartbeat is also running and every time the heartbeat is called, it is setting the internal schedulable to true. Since we don't want to kill the virt-handler, this prevents the heartbeat from setting nodes to schedulable=true when they should not be schedulable.